config/share_notify.yml

```diff
@@ -1,3 +1,30 @@
----
-host: <%= ENV.fetch('APPS_DHH_STAGING_SHARE_NOTIFY_HOST') {'https://staging.osf.io'}%>
-token: <%= ENV.fetch('APPS_DHH_STAGING_NOTIFY_TOKEN') {'default'}%>
\ No newline at end of file
+defaults: &defaults
+gimili:
+  host: <%= ENV.fetch('APPS_DHH_STAGING_SHARE_NOTIFY_HOST') {'https://staging.osf.io/'}%>
+  token: <%= ENV.fetch('APPS_DHH_STAGING_NOTIFY_TOKEN') {'share_token'}%>
+
+development:
+  <<: *defaults
+  
+pontos:
+  <<: *defaults
+
+gimili:
+  <<: *defaults
+
+staging:
+  <<: *defaults
+
+qa:
+  <<: *defaults
+
+demo:
+  <<: *defaults
+
+test:
+  <<: *defaults
+
+production:
+  host: <%= ENV.fetch('APPS_DHH_PROD_SHARE_NOTIFY_HOST') {'https://staging.osf.io/'}%>
```

Comment:
do we want production to fallback to staging? this seems potentially problematic. I wonder, in the scenario where production doesn't have the proper environment variables, if it wouldn't be better to have it fail with a KeyError so we know something is wrong?